src/com/google/ProfilingReport.java

```diff
@@ -0,0 +1,249 @@
+   * where executed to get aggregated instrumentation data.
+   */
+  private static float calculatePercentOfInstrumentationExecuted(
+      List<Map<String, List<ProfilingResult>>> result,
+      Predicate<InstrumentationType> instrumentationTypeCheck) {
+    int totalInstrumentationPoints = 0;
+    int totalInstrumentationPointsExecuted = 0;
+
+    for (Map<String, List<ProfilingResult>> fileProfilingResult : result) {
+      for (String key : fileProfilingResult.keySet()) {
+        List<ProfilingResult> listOfProfilingResult = fileProfilingResult.get(key);
+        for (ProfilingResult profilingResult : listOfProfilingResult) {
+          if (instrumentationTypeCheck.test(profilingResult.type())) {
+            totalInstrumentationPoints++;
+            if (profilingResult.executed() != 0) {
+              totalInstrumentationPointsExecuted++;
+            }
+          }
+        }
+      }
+    }
+    return roundFloat2Decimals(
+        (float) totalInstrumentationPointsExecuted / totalInstrumentationPoints);
+  }
+
+  private void calculateProfilingReportPercents() {
+
+    List<Map<String, List<ProfilingResult>>> listOfProfilingResultMaps = new ArrayList<>();
+    for (FileProfilingResult fileProfilingResult : result) {
+      listOfProfilingResultMaps.add(fileProfilingResult.profilingDataPerFunction());
+    }
+
+    this.percentOfFunctionsExecuted = calculatePercentOfInstrumentationExecuted(
+        listOfProfilingResultMaps, (t) -> t.equals(InstrumentationType.FUNCTION));
+    this.percentOfBranchesExecuted = calculatePercentOfInstrumentationExecuted(
+        listOfProfilingResultMaps, (t) -> t.equals(InstrumentationType.BRANCH) || t
+            .equals(InstrumentationType.BRANCH_DEFAULT));
+
+  }
+
+  /**
+   * A class which groups profiling results by source file and contains a mapping of them to each
+   * function which is present in that source file. Used also for Json conversion.
+   */
+  @AutoValue
+  abstract static class FileProfilingResult {
+
+    static FileProfilingResult create(String fileName, float percentOfFunctionsExecuted,
+        float percentOfBranchesExecuted,
+        Map<String, List<ProfilingResult>> profilingDataPerFunction) {
+      return new AutoValue_ProfilingReport_FileProfilingResult(fileName, percentOfFunctionsExecuted,
+          percentOfBranchesExecuted, profilingDataPerFunction);
+    }
+
+    abstract String fileName();
+
+    abstract float percentOfFunctionsExecuted();
+
+    abstract float percentOfBranchesExecuted();
+
+    abstract Map<String, List<ProfilingResult>> profilingDataPerFunction();
+  }
+
+  /**
+   * This class contains a detailed breakdown of each instrumentation point. For any encoded param,
+   * this class will contain its details. The `executed` property will be a percent of total times
+   * the instrumentation point was called across all provided instrumentation reports. i.e if
```

Comment:
very nitpicky: `i.e.` should be `e.g.` here

https://www.proofreadnow.com/blog/ie-vs-eg-and-when-to-use-which

also, I think it would make more sense to document `executed` above the `executed()` getter, instead of here.